server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -436,6 +440,13 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
                     throw new IllegalArgumentException("Cannot have more than one plugin implementing a REST wrapper");
                 }
                 restWrapper = newRestWrapper;
+                if (restWrapper.getClass().getCanonicalName() == null ||
+                    restWrapper.getClass().getCanonicalName().startsWith("org.elasticsearch") == false) {
+                    deprecationLogger.deprecate(DeprecationCategory.SECURITY, "3rd_party_sec_deprecation", "Use of third-party " +
+                        "security plugins is deprecated and will not be possible in 8.0. You can use built-in Elasticsearch features " +
```

Comment:
I am concerned that the message is exclusively about "third party security" when it's only detecting a rest wrapper.

I think we need to be more clear than that. Anyone who is using the plugin API to install a rest wrapper is going to be affected, and we need to be clear about that.
 
I would propose a message like:

> The {plugin-name} plugin installs a custom REST wrapper. This functionality is deprecated and will not be possible in Elasticsearch 8.0. If this plugin is intended to provide security features for Elasticsearch then you should switch to using the built-in Elasticsearch features instead.